app/frontend/components/shared/permit-applications/scroll-link.tsx

```diff
@@ -1,16 +1,39 @@
 import { Link } from "@chakra-ui/react"
 import React from "react"
 
-export const ScrollLink = ({ to, children, ...props }) => {
-  const handleClick = (event) => {
+interface ScrollLinkProps {
+  to: string
+  children: React.ReactNode
+  [key: string]: any
+}
+
+export const ScrollLink: React.FC<ScrollLinkProps> = ({ to, children, ...props }) => {
+  const handleClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
     event.preventDefault()
-    const targetElement = document.getElementById(to)
+    let targetElement = document.getElementById(to)
+    // If no element found by ID, try finding by class
+    if (!targetElement) {
+      const classElements = document.getElementsByClassName(to)
+      if (classElements.length > 0) {
+        targetElement = classElements[0] as HTMLElement
+      }
+    }
     if (targetElement) {
+      // Scroll to the target element
       targetElement.scrollIntoView({ behavior: "smooth", block: "center" })
-      const id = targetElement.querySelector("label.col-form-label")?.getAttribute("for")
-      const formControlElement = document.getElementById(id)
 
+      // Focus the target element
+      targetElement.tabIndex = -1 // Ensure the element can be focused
+      targetElement.focus()
+
+      // Focus on the target form control if present
+      const id = targetElement.querySelector("label.col-form-label")?.getAttribute("for")
+      const formControlElement = id ? document.getElementById(id) : null
       formControlElement?.focus?.()
+
+      targetElement.addEventListener("blur", () => {
+        targetElement.removeAttribute("tabIndex") // Removes the tabIndex attribute, returning it to its default behavior
+      })
     }
```

Comment:
This will create a lot of event listeners as for every click a new listener is attached. You will have to look into it, but find a way also cleanup listeners as this might cause memory issues. I quickly chat gpted it https://chatgpt.com/share/811f6752-d68f-4b8e-9df4-c3480bc00fdf, and the solution seems simple enough but I did not look too deeply into it